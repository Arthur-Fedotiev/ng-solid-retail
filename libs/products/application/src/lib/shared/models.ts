import { Categories, Colors, Retailers, Tiers } from './constants';

export type CategoriesMap = typeof Categories;
export type Category = typeof Categories[keyof typeof Categories];

export type RetailersMap = typeof Retailers;
export type TiersMap = typeof Tiers;
export type ColorsMap = typeof Colors;
