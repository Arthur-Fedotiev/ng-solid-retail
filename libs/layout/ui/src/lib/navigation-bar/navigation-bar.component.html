<nav class="nav-container">
  <mat-nav-list class="nav-container__list">
    <ng-content></ng-content>
  </mat-nav-list>
</nav>
