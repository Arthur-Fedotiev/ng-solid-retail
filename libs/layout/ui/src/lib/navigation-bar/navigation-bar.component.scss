.nav-container {
  width: 13rem;
}
