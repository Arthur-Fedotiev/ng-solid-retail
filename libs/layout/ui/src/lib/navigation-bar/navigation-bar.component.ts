import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';

@Component({
  selector: 'omnia-navigation-bar',
  templateUrl: './navigation-bar.component.html',
  styleUrls: ['./navigation-bar.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class NavigationBarComponent implements OnInit {
  constructor() {}

  ngOnInit(): void {}
}
