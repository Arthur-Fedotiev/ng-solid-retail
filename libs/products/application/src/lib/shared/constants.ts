import {
  ProductCategory,
  ProductRetailer,
  ProductTier,
  SHOES_AND_CLOTHING_COLORS,
} from '@sr/products/entities';

export const Categories = ProductCategory;
export const Retailers = ProductRetailer;
export const Tiers = ProductTier;
export const Colors = {
  [Categories.Shoes]: SHOES_AND_CLOTHING_COLORS,
  [Categories.Clothing]: SHOES_AND_CLOTHING_COLORS,
} as const;
