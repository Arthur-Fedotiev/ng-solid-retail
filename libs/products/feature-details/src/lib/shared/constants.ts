import { Tiers } from '@sr/products/application';

export const PRODUCT_TIERS = Tiers;
