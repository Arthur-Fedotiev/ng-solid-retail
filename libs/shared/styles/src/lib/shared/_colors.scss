$colors: (
  'M900': #1b1c1d,
  'M800': #4e5052,
  'M700': #969798,
  'M600': #f1f3f4,
  'M500': #727779,
  'M300': #d9dadc,
  'M100': #f1f3f4,
  'M0': #ffffff,
  'S50': #fafdfd,
  'S100': #f0f8f8,
  'B900': #4b79f7,
  'R50': #ffebee,
  'R100': #ffcacf,
  'R500': #fa3126,
  'R600': #eb2327,
  'R700': #d91421,
);

@function app-color($color) {
  @return map-get($colors, $color);
}
