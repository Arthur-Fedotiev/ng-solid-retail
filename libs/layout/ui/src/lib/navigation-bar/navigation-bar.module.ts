import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NavigationBarComponent } from './navigation-bar.component';
import { SharedUiMaterialModule } from '@sr/shared/ui-material';

@NgModule({
  declarations: [NavigationBarComponent],
  imports: [CommonModule, SharedUiMaterialModule],
  exports: [NavigationBarComponent],
})
export class NavigationBarModule {}
