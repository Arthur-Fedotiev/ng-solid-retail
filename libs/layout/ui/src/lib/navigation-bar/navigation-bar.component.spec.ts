import { ComponentFixture, TestBed } from '@angular/core/testing';
import { SharedUiMaterialModule } from '@omnia/shared/ui-material';

import { NavigationBarComponent } from './navigation-bar.component';

describe('NavigationBarComponent', () => {
  let component: NavigationBarComponent;
  let fixture: ComponentFixture<NavigationBarComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [SharedUiMaterialModule],
      declarations: [NavigationBarComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(NavigationBarComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
