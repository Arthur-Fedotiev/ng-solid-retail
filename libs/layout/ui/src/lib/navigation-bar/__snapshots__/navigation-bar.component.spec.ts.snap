// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBarComponent should render 1`] = `
<sr-navigation-bar
  __ngContext__={[Function Number]}
>
  <nav
    class="nav-container"
  >
    <mat-nav-list
      aria-disabled="false"
      class="mat-mdc-nav-list mat-mdc-list-base mdc-list nav-container__list"
      role="navigation"
    />
  </nav>
</sr-navigation-bar>
`;
